desktop/tests/test_evals.py

```
@@ -0,0 +1,1178 @@
+    assert candidate_run.passed
+    assert report.comparisons[0].correctness.candidate_pass_rate == 1.0
+    assert report.comparisons[0].judge_scores.observational is True
+    assert report.comparisons[0].judge_scores.candidate_mean == 0.1
+    assert report.comparisons[0].judge_scores.mean_delta == pytest.approx(-0.8)
+
+
+@pytest.mark.parametrize("score", [float("nan"), float("inf"), float("-inf"), -0.1, 1.1])
+def test_judge_observation_requires_a_finite_unit_score(score):
+    from coworker.evals.models import JudgeObservation
+
+    with pytest.raises(ValueError, match="finite.*0.*1"):
```

Ruff (0.16.2)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)